### PR TITLE
11 get articles query topic

### PR DESCRIPTION
**Pull request by JoWatson2011**
I've added the functionality to query the /api/articles endpoint by topic. 
I've also added a new util function to generically query tables for keys.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -154,7 +154,14 @@ describe("GET /api/articles", () => {
         });
       });
   });
-  test.todo("400: Responds with Bad Request when topic query is not valid");
+  test("404: Responds with Not Found when topic query is not valid - i.e. not found in topic table", () => {
```
Comment (nc-aimee-muter) [inline __tests__/app.test.js:157]: Nice error test.

**Review by nc-aimee-muter — COMMENTED** [review]
Looking good! I'm not sure if I've just missed it but we need a 200 test to check that when there are no articles for a queried topic where the topic DOES exist, we get an empty array.
